components/blocks/cards/SignUpCard/SignUpCard.vue

```diff
@@ -204,27 +179,30 @@ async function signup() {
                   ? 'rgb(185 28 28 / 1)'
                   : '',
               }"
-              :type="state.showPassword.value ? 'text' : 'password'"
+              :show-password="state.showPassword.value"
               placeholder="Confirm"
               autocomplete="password"
               data-testid="password-confirm-input"
               minlength="8"
               maxlength="80"
               @change="
                 passwordConfirmValid =
-                  passwordsAreTheSame() && isPasswordValid()
+                  passwordsAreTheSame(
+                    register.password,
+                    register.passwordConfirm,
+                  ) && isPasswordValid(register.password)
               "
+              @hide-show-clicked="toggleShowPassword"
             />
 
             <HideShowPassword
               id="hide-show-password"
               type="button"
               data-testid="hide-show-button"
-              @click="state.showPassword.value = !state.showPassword.value"
-              @keydown.enter.prevent=""
-              @keyup.enter="
-                state.showPassword.value = !state.showPassword.value
-              "
+              :hidden="state.showPassword.value"
+              @click="toggleShowPassword"
+              @keydown.enter="$event.preventDefault()"
```

Comment:
just curious as to the cahnge on this line as the previous does the same thing in a bit less lines. Not a big deal of course tho just curious

Reply:
Just made the change so that way we don't have the same bit written a bunch of times. Then if we ever need to do anything else in here, it'll apply elsewhere. Basically just drying things up